### DISCUSSION TITLE
Any plans for Godot 4?

### DISCUSSION THREAD

**Discussion by rosshadden** [imjp94/gd-YAFSM#42] [Q&A]
I want to open a discussion about Godot 4.

Do you have anything in the works already or a plan to support Godot 4?

I'm not sure how this would even look to be honest because it wouldn't be backwards compatible, so I'm not sure what the best way to go about it would be. But fortunately I would imagine it's a common situation in the community.

**Comment by piratesephiroth** [comment]
A great part of the plugin functionality is incomplete in Godot 4 right now.

**Comment by imjp94** [comment]
I haven't start working on it yet as I am busy with my own game dev, so if anyone would love to contribute can submit a PR.

For the development, here's how I would handle it:
- `master` branch always support for the latest stable Godot version, which is 3.x by now, while Godot 4 will be worked in the branch `godot4`
- When Godot 4 stable released, 
  - branch `godot3` will be created for the ease of bug fixing
  - branch `godot4` will be merged to `master` and tagged as `v1.0` due to the breaking changes

**Comment by imjp94** [comment]
@adamscott is currently working on porting to Godot 4.0, check out his works if you needs to use gd-YAFSM in Godot 4.0

**Comment by adamscott** [comment]
I just submitted a WIP PR. #47 

There's some issues, but I managed to make the library work with a simple state machine. Feel free to fork it and test it out.

**Comment by jbreitweiser** [comment]
Since we are currently on Godot 4.4 are there plans to update the AssetLib with the latest release.  It still pulls down a broken release.  Same question for the demo repo which does not currently function?

**Reply by imjp94** [reply]
Thanks for reminding, I have submit new edit to Godot Asset Library
